src/librustc_typeck/constrained_type_params.rs

```diff
@@ -13,9 +13,14 @@ use rustc::ty::fold::{TypeFoldable, TypeVisitor};
 use rustc::util::nodemap::FnvHashSet;
 
 #[derive(Clone, PartialEq, Eq, Hash, Debug)]
-pub enum Parameter {
-    Type(ty::ParamTy),
-    Region(ty::EarlyBoundRegion),
+pub struct Parameter(pub u32);
```

Comment:
Can you please leave a comment explaining what this field is? 

Also, I think the field should be private, and you should write `fn from_index(usize) -> Parameter`; it's always a shame when the public API doesn't expose `usize`, I think.